### ISSUE TITLE
Implementing Display for error

### ISSUE THREAD

**Issue by vmx** [supranational/sppark#10]
Currently the cuda errors [`impl From<Error> for String`](https://github.com/supranational/sppark/blob/da968db503e05b77236eb3ba78ec51e6a1075f24/rust/src/lib.rs#L27). If instead it would implement `Display` (see https://rust-lang.github.io/api-guidelines/interoperability.html?highlight=display#error-types-are-meaningful-and-well-behaved-c-good-err for more information), you could e.g. also call `err.to_string()` or even use it like `println!("error was : {}", err)`.

If I should provide a pull request, let me know.

**Comment by dot-asm** [comment]
> If instead ...

Does it have to be "instead"? I mean what prevents us from just **adding** it? Not that I have some attachment to From, it's just that yanking it is likely to be considered bad style. You kind of add something new and remove old later...

> If I should provide a pull request, let me know.

If it's **just** Display, I can handle it. But maybe there is more to it. Say it might be appropriate to implement [Error](https://doc.rust-lang.org/std/error/trait.Error.html) trait. In other words if you can think of something more nuanced than adding Display, then a pull request would be appreciated. Well, it would be appreciated in either case, but we do need our Rust exercises too, you know :-)

**Comment by dot-asm** [comment]
I've added Display, simplified for now as Error needs an overhaul. Because we've discussed using it for conveying errors other than CUDA.

> One thing I don't fully understand is, why the CUDA error is wrappen in a macro.

The original rationale was that we don't know if it will be needed (because target application might not be using CUDA code), and providing a macro would give the application freedom to decide. But since it's discussed to be used for other errors, it's arguably appropriate to change that...
